### PR TITLE
Fix the utf8 convertor for VS2017

### PR DESCRIPTION
**Pull request by shravanrn**
The UTF8 conversion functions are broken in a slightly different way in VS2017. Add functions to hide these differences

### CODE REVIEW

**Comment by shravanrn** [comment]
The errors in the CI build process seem to be unrelated to this change

File: runtime/Cpp/runtime/src/support/StringUtils.h [hunk 1]
```diff
@@ -10,12 +10,34 @@
 namespace antlrcpp {
   // For all conversions utf8 <-> utf32.
   // VS 2015 has a bug in std::codecvt_utf8<char32_t> (VS 2013 works fine).
-#if defined(_MSC_VER) && _MSC_VER == 1900
+#if defined(_MSC_VER) && _MSC_VER > 1900 && _MSC_VER < 2000
```
Comment (KindDragon) [inline runtime/Cpp/runtime/src/support/StringUtils.h]: Shouldn't be `#if defined(_MSC_VER) && _MSC_VER >= 1900`?

Comment (mike-lischke) [inline runtime/Cpp/runtime/src/support/StringUtils.h]: Yes, true. @shravanrn can you fix that yet?

Comment (KindDragon) [inline runtime/Cpp/runtime/src/support/StringUtils.h]: and bellow

**Comment by shravanrn** [comment]
Sorry about the delay. Based on comments I realized I had actually gotten this slightly incorrect the last. I have double checked that the following works for both VS2015 and VS 2017

**Comment by shravanrn** [comment]
@mike-lischke @KindDragon - Just a friendly reminder :)

**Comment by mike-lischke** [comment]
We need @parrt for a merge. Development on ANTLR4 stopped abruptly and everybody is now waiting for a new time slot where things begin to move again. However, it's not clear that "everybody" has time to move at the same moment as the ANTLR4 starts moving again...

**Comment by parrt** [comment]
Sorry...here ya go.

**Pull request merged**